docs/pages/web/utilities/devicetypeprovider.js

```diff
@@ -1,22 +1,50 @@
 // @flow strict
 import { type Node } from 'react';
+import { SlimBanner } from 'gestalt';
 import GeneratedPropTable from '../../../docs-components/GeneratedPropTable.js';
-// To be used when this page is made live
-// import MainSection from '../../../docs-components/MainSection.js';
+import MainSection from '../../../docs-components/MainSection.js';
 import Page from '../../../docs-components/Page.js';
 import PageHeader from '../../../docs-components/PageHeader.js';
 import docgen, { type DocGen } from '../../../docs-components/docgen.js';
-
-// TODO: @rjames
-// - Add usage examples to this docs page
-// - Add reference to this docs page in docs/docs-components/sidebarIndex.js
+import SandpackExample from '../../../docs-components/SandpackExample.js';
+import implementation from '../../../examples/devicetypeprovider/implementation.js';
 
 export default function DocsPage({ generatedDocGen }: {| generatedDocGen: DocGen |}): Node {
   return (
-    <Page title="DeviceTypeProvider">
-      <PageHeader name="DeviceTypeProvider" description={generatedDocGen?.description} />
+    <Page title={generatedDocGen?.displayName}>
+      <PageHeader name={generatedDocGen?.displayName} description={generatedDocGen?.description} />
+
+      <SlimBanner
+        type="info"
+        iconAccessibilityLabel="Recommendation"
+        message="Gestalt components that require DeviceTypeProvider to enable their mobile user interfaces have a Mobile variant section in their documentation page. Check each example to learn more about their particular implementations."
+      />
 
       <GeneratedPropTable generatedDocGen={generatedDocGen} />
+
+      <MainSection
+        name="Implementation"
+        description={`Gestalt components can have different user interfaces based on the device platform they are rendered: desktop or mobile.
```

Comment:
Gestalt components can have different interfaces depending on the user's device. We currently support `'desktop'` and `'mobile'`. Components default to a responsive "desktop" UI. DeviceTypeProvider is required to enable mobile-specific variants where available. This provider should be implemented at the top level of your app. Any additional nested DeviceTypeProviders will override the top-level configuration.

While device detection can (for now) be performed using the [user-agent string](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent), this is [not generally recommended](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent). Thankfully, there are [better solutions](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent#avoiding_user_agent_detection)!

This example shows a component with different desktop and mobile UIs.

Reply:
⬆️ meant as a suggestion for this text block

Reply:
### DONE